skbio/io/format/embl.py

```diff
@@ -0,0 +1,1497 @@
+from skbio.io.format._sequence_feature_vocabulary import (
+    _yield_section, _parse_single_feature, _serialize_section_default,
+    _serialize_single_feature)
+from skbio.metadata import IntervalMetadata
+from skbio.sequence import Sequence, DNA, RNA, Protein
+from skbio.util._misc import chunk_str
+
+
+# look at skbio.io.registry to have an idea on how to define this class
+embl = create_format('embl')
+
+# This list is ordered used to read and write embl file. By processing those
+# values one by one, I will write embl sections with the same order
+_HEADERS = [
+    'LOCUS',
+    'ACCESSION',
+    'PARENT_ACCESSION',
+    'PROJECT_IDENTIFIER',
+    'DATE',
+    'DEFINITION',
+    'GENE_NAME',
+    'KEYWORDS',
+    'SOURCE',
+    'REFERENCE',
+    'DBSOURCE',
+    'COMMENT',
+    'FEATURES'
+    ]
+
+
+# embl has a series of keys different from genbank; moreover keys are not so
+# easy to understand (eg. RA for AUTHORS). I want to use the same keys used by
+# genbank both to convert between formats and to use the same methods to get
+# info from Sequence and its derived objects Here is a dictionary of keys
+# conversion (EMBL->GB). All the unspecified keys will remain in embl format
+KEYS_TRANSLATOR = {
+                   # identification
+                   'ID': 'LOCUS',
+                   'AC': 'ACCESSION',
+                   # PA means PARENT ACCESSION (?) and applies to
+                   # feature-level-products entries
+                   'PA': 'PARENT_ACCESSION',
+                   'PR': 'PROJECT_IDENTIFIER',
+                   'DT': 'DATE',
+                   'DE': 'DEFINITION',
+                   'GN': 'GENE_NAME',  # uniprot specific
+                   'KW': 'KEYWORDS',
+                   # Source (taxonomy and classification)
+                   'OS': 'ORGANISM',
+                   'OC': 'taxonomy',
+                   'OG': 'organelle',
+                   # reference keys
+                   'RA': 'AUTHORS',
+                   'RP': 'REFERENCE',
+                   'RC': 'REFERENCE_COMMENT',
+                   'RX': 'CROSS_REFERENCE',
+                   'RG': 'GROUP',
+                   'RT': 'TITLE',
+                   'RL': 'JOURNAL',
+                   # Cross references
+                   'DR': 'DBSOURCE',
+                   'CC': 'COMMENT',
+                   # features
+                   'FH': 'FEATURES',
+                   'FT': 'FEATURES',
+                   'SQ': 'ORIGIN',
+                   }
+
+# the original genbank _yield_section divides entries in sections relying on
+# spaces (the same section has the same level of indentation). Uniprot entries
+# have a key for each line, so to divide record in sections I need to define a
+# corresponance for each key to section, then I will divide a record in
+# sections using these section name. Some keys are commented out since I don't
+# find them in example. What I have to do if I find them?
```

Comment:
I agree. When I wrote this comment, there are keys that I can't handle. Non I can handle almost any of such keys. The last key I added is 'CO' that stand for constructed sequences like [LT357133](http://www.ebi.ac.uk/ena/data/view/LT357133&display=text). Such entries don't have sequences (so no intervalmetadata is possible) but but *rather the assembly information on all accession.versions and sequence locations relevant to building the constructed sequence*. This is an example of referencing external sequences. At the moment, I read data (metadata) in order to not raise an exception